meta.go

```diff
@@ -189,14 +194,21 @@ func NewDecoder(destStruct interface{}) *Decoder {
 				dfield.fieldCategory = categoryAllFieldsMap
 			} else if valuer, ok := fieldInterface.(Valuer); ok {
 				dfield.fieldCategory = categoryValuer
-				dfield.Options = valuer.ParseOptions(fieldStruct.Tag)
+
+				parsedOptions := valuer.ParseOptions(fieldStruct.Tag)
+				if timeOptions, ok := parsedOptions.(*TimeOptions); ok && len(options.TimeFormats) > 0 {
```

Comment:
It would be nice if this rule could be implemented in `time.go` rather than here, since it is specific to that type. Maybe the `TimeOptions` type could implement an interface with 1 function `ApplyDecoderOptions`, or something.